ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandStoreTest.java

```diff
@@ -99,8 +107,160 @@ public void shouldHaveAllCreateCommandsInOrder() {
         new Pair<>(createId, latestCommand))));
   }
 
+  private CommandIdAssigner buildSameIdAssigner() {
+    final CommandIdAssigner commandIdAssigner = mock(CommandIdAssigner.class);
+    expect(commandIdAssigner.getCommandId(anyObject())).andStubAnswer(
+        () -> new CommandId(CommandId.Type.STREAM, "foo", CommandId.Action.CREATE)
+    );
+    replay(commandIdAssigner);
+    return commandIdAssigner;
+  }
+
+  private void setupConsumerToReturnCommand(final CommandId commandId, final Command command) {
+    reset(commandConsumer);
+    expect(commandConsumer.poll(anyObject(Duration.class))).andReturn(
+        new ConsumerRecords<>(
+            Collections.singletonMap(
+                new TopicPartition("command-topic", 0),
+                Collections.singletonList(
+                    new ConsumerRecord<>(
+                        "command-topic", 0, 0, commandId, command)
+                ))
+        )
+    ).times(1);
+    replay(commandConsumer);
+  }
+
+  @Test
+  public void shouldFailEnqueueIfCommandWithSameIdRegistered() {
+    final String statementText = "test-statement";
+
+    final Statement statement = mock(Statement.class);
```

Comment:
Still a lot of common code shared between these test cases. 

- Can we not move the mock creation etc into fields?
- `commandIdAssigner` is only used to pass to `createCommandStore`.  Why not just call `buildSameIdAssigner` from within `createCommandStorez`?